sbi/neural_nets/classifier.py

```diff
@@ -1,50 +1,28 @@
 # This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
 # under the Affero General Public License v3, see <https://www.gnu.org/licenses/>.
 
+from __future__ import annotations
+
 from typing import Optional
 
-import torch
 from pyknos.nflows.nn import nets
 from torch import Tensor, nn, relu
 
+from sbi.neural_nets.ratio_estimators import TensorRatioEstimator
 from sbi.utils.sbiutils import standardizing_net, z_score_parser
 from sbi.utils.user_input_checks import check_data_device, check_embedding_net_device
 
 
-class StandardizeInputs(nn.Module):
-    def __init__(self, embedding_net_x, embedding_net_y, dim_x, dim_y):
-        super().__init__()
-        self.embedding_net_x = embedding_net_x
-        self.embedding_net_y = embedding_net_y
-        self.dim_x = dim_x
-        self.dim_y = dim_y
-
-    def forward(self, inputs: list) -> Tensor:
-        assert (
-            isinstance(inputs, list) and len(inputs) == 2
-        ), """Inputs to (s)nre classifier must be a list containing raw theta and x."""
-        out = torch.cat(
-            [
-                self.embedding_net_x(inputs[0]),
-                self.embedding_net_y(inputs[1]),
-            ],
-            dim=1,
-        )
-        return out
-
-
-def build_input_layer(
+def build_z_scored_embedding_nets(
```

Comment:
This feels a bit specific, why not do:
```
def build_z_scored_embedding_nets(
    batch,
    z_score, 
    embedding_net
):
    ...
    return embedding_net
```

This way you can call this as many time as you have embedding nets?

Reply:
good idea